dask/dataframe/io/parquet.py

```diff
@@ -309,6 +305,11 @@ def _read_fastparquet(fs, fs_token, paths, columns=None, filters=None,
                        categories, pf.schema, pf.cats, pf.dtypes,
                        pf.file_scheme, storage_name_mapping)
            for i, rg in enumerate(rgs)}
+    for catcol in pf.cats:
+        if catcol in meta.columns:
+            meta[catcol] = meta[catcol].cat.set_categories(pf.cats[catcol])
+        if meta.index.name == catcol:
```

Comment:
Can both branches here ever be taken (I don't think so?). If not, should use `elif`.

Reply:
Also, can you move this up to where `meta` is created? Otherwise there's a bit of space in between where `meta` exists but is incorrect for the given frame.

I also believe this currently fails if the output is a Series, as `meta` at this point would have no `.columns` attribute. Would be good to add a test for this as well.